src/lib/salad/stailq.h

```diff
@@ -67,8 +75,16 @@ inline static void
 stailq_add(struct stailq *head, struct stailq_entry *item)
 {
 	item->next = head->first;
-	if (item->next == NULL)
+	/*
+	 * Without curly braces gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
+	 * generates code such that statement is executed unconditionally.
+	 */
+	if (item->next == NULL) {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Waddress-of-packed-member"
 		head->last = &item->next;
+#pragma GCC diagnostic pop
```

Comment:
Why not use memset here? The pragmas may not work with other compilers (e.g. clang).

Reply:
After discussion with @locker and @alyapunov we decided to try approach without `memcpy`. So it is not applicable anymore.